relay-event-normalization/src/normalize/mod.rs

```diff
@@ -966,10 +956,19 @@ impl<'a> DynamicMeasurementsConfig<'a> {
     /// Returns an iterator over the union of [`BuiltinMeasurementKey`]s from both the global and
     /// project-level [`MeasurementsConfig`]s, with the elements from project config coming first.
     pub fn builtin_measurement_keys(&'a self) -> impl Iterator<Item = &'a BuiltinMeasurementKey> {
-        let from_project = self.project.iter().flat_map(|c| &c.builtin_measurements);
-        let from_global = self.global.iter().flat_map(|c| &c.builtin_measurements);
+        let from_project = self
+            .project
+            .iter()
+            .flat_map(|c| &c.builtin_measurements)
+            .collect::<Vec<_>>();
+        let from_project_clone = from_project.clone();
```

Comment:
Do we need two vectors? The `contains` below is O(n), could we have an iterator and a set? We could use the set for checking duplicates and the iterator to chain the other one.

Reply:
I did try with a set but ran into many problems, and it's also not clear that its actually better. the amount of elements are likely quite small so the overhead of a hashset might be more significant than the time it takes to iterate through it. perhaps it's different with a btreeset

Reply:
This seems to work:

```rust
    pub fn builtin_measurement_keys(&'a self) -> impl Iterator<Item = &'a BuiltinMeasurementKey> {
        let from_project = self.project.iter().flat_map(|c| &c.builtin_measurements);
        let mut project_set = HashSet::<&BuiltinMeasurementKey>::new();
        project_set.extend(from_project.clone());

        let from_global = self
            .global
            .iter()
            .flat_map(|c| &c.builtin_measurements)
            .filter(move |&k| !project_set.contains(k));

        from_project.into_iter().chain(from_global)
    }
```

Reply:
Resolving this comment as the size of the collection is not expected to be high enough for the performance benefit.